syft/messaging/plan.py

```diff
@@ -74,210 +65,330 @@ class State(object):
     sure they are provided to remote workers who are sent the Plan.
     """
 
-    def __init__(self, plan):
-        self.keys = set()
+    def __init__(self, owner, plan=None, keys=None, state_ids=None):
+        self.owner = owner
         self.plan = plan
+        self.keys = keys or set()
```

Comment:
Are keys still used? If so, what is the difference between them and state_ids?

Reply:
Nope I forgot to remove the last occurences, thank you :)